### PR TITLE
OSDOCS-5420: Auto generate oc commands for MicroShift

### PR DESCRIPTION
**Pull request by ShaunaDiaz**
Version(s):
4.13 only

Issue:
https://issues.redhat.com/browse/OSDOCS-5420

Link to docs preview:
https://57231--docspreview.netlify.app/microshift/latest/microshift_cli_ref/microshift-oc-cli-commands-list.html 

QE review:
- [x] QE has approved this change.

### CODE REVIEW

**Comment by dhellmann** [comment]
This list looks OK as a first pass. It includes `autoscale`, which we don't want, and it includes a bunch of the `oc adm` commands because I failed to notice those were in a separate documentation file when I was building my list of commands to review when giving @soltysh a list to work from.

I suggest we take this as it is and refine it in a follow-up.

/lgtm

**Comment by kalexand-rh** [comment]
@dhellmann, is the current set of commands correct?

**Comment by kalexand-rh** [comment]
@ShaunaDiaz, will you send this through QE for a +1 before we merge?

**Comment by ShaunaDiaz** [comment]
@kalexand-rh Is that step typically done for OCP? I didn't think so since these are auto-generated. Figured QE was done before the workloads team releases them. Is that not the case?

**Comment by ShaunaDiaz** [comment]
@jogeo or @dhensel-rh Could one of you PTAL at the oc commands here for MicroShift per @kalexand-rh's comment above and see if there is anything concerning? I can easily edit out commands as needed. Give it an lgtm if these look OK, please?

File: modules/microshift-oc-by-example-content-gen.adoc [hunk 1]
```diff
@@ -0,0 +1,2385 @@
...
+  # Get output from running the 'date' command in ruby-container from pod mypod
+  oc exec mypod -c ruby-container -- date
+
+  # Switch to raw terminal mode; sends stdin to 'bash' in ruby-container from pod mypod
+  # and sends stdout/stderr from 'bash' back to the client
+  oc exec mypod -c ruby-container -i -t -- bash -il
+
+  # List contents of /usr from the first container of pod mypod and sort by modification time
+  # If the command you want to execute in the pod has any flags in common (e.g. -i),
+  # you must use two dashes (--) to separate your command's flags/arguments
```
[the remark this answers is not in the document]
Comment (GroceryBoyJr) [reply modules/microshift-oc-by-example-content-gen.adoc:1245]: This suggestion is more of a question: won't using "/" to mean "or" confuse people who think of "|" as the OR symbol, and think of "/" as a character in an option or a filename?

**Review by GroceryBoyJr — COMMENTED** [review]
@ShaunaDiaz I'm done, just a few small nits. Please send back through peer review after QE gets a look at it. You can ping me on it if you like since I'm now somewhat familiar.

File: modules/microshift-oc-adm-by-example-content.adoc [hunk 2]
```diff
@@ -0,0 +1,751 @@
...
+  oc adm migrate template-instances
+
+  # To actually perform the update, the confirm flag must be appended
+  oc adm migrate template-instances --confirm
+----
+
+
+
+== oc adm must-gather
+Launch a new instance of a pod for gathering debug information
```
[the remark this answers is not in the document]
Comment (ShaunaDiaz) [reply modules/microshift-oc-adm-by-example-content.adoc:263]: @GroceryBoyJr It doesn't look like we edit these auto-generated files at all: https://docs.openshift.com/container-platform/4.12/cli_reference/openshift_cli/administrator-cli-commands.html#oc-adm-must-gather

[the remark this answers is not in the document]
Comment (ShaunaDiaz) [reply modules/microshift-oc-adm-by-example-content.adoc:263]: @kalexand-rh Once set up properly, do these auto-generated `oc` files usually go through peer review? Or is there an abbreviated process for these?

**Comment by dhensel-rh** [comment]
There are several references to 4.11.2 in modules/microshift-oc-adm-by-example-content.adoc. 

Is it possible to update the document to reference 4.13. I don't think MicroShift was available prior to 4.12 ?

**Comment by ShaunaDiaz** [comment]
@dhensel-rh MicroShift wasn't, but OCP was. Since this is the OCP CLI tool that happens to work in MicroShift, I think it's OK. They are just meant as examples for how to use the command. The OCP version of the same files leaves in those 4.11 version numbers, too. @dhellmann can you confirm/deny what I am thinking here?

**Comment by dhellmann** [comment]
> @dhensel-rh MicroShift wasn't, but OCP was. Since this is the OCP CLI tool that happens to work in MicroShift, I think it's OK. They are just meant as examples for how to use the command. The OCP version of the same files leaves in those 4.11 version numbers, too. @dhellmann can you confirm/deny what I am thinking here?

Yes, that's fine, we don't need to rewrite the examples in `oc` for those commands.

**Pull request merged**